dll/ntdll/ldr/ldrapi.c

```diff
@@ -1161,28 +1164,23 @@ LdrEnumerateLoadedModules(IN BOOLEAN ReservedFlag,
         /* Break if we were asked to stop enumeration */
         if (Stop)
         {
-            /* Release loader lock */
-            Status = LdrUnlockLoaderLock(0, Cookie);
-
-            /* Reset any successful status to STATUS_SUCCESS, but leave
-               failure to the caller */
-            if (NT_SUCCESS(Status))
-                Status = STATUS_SUCCESS;
-
-            /* Return any possible failure status */
-            return Status;
+            break;
         }
 
         /* Advance to the next module */
         ListEntry = ListEntry->Flink;
     }
 
-    /* Release loader lock, it must succeed this time */
+    /* Release loader lock */
     Status = LdrUnlockLoaderLock(0, Cookie);
-    ASSERT(NT_SUCCESS(Status));
```

Comment:
I am not sure that removing this assert is a fix.

Reply:
Committed. Note: After investigation, Windows won't break here, but `LDR_UNLOCK_LOADER_LOCK_FLAG_RAISE_ON_ERRORS` didn't specify to `LdrUnlockLoaderLock`, so assert here is make sense.

Reply:
This `ASSERT()` could be moved to an `else`, now that there is an `if`.

Reply:
Whatever the callback stop enumeration or not, the following actions should be the same, just unlock the loader lock and return status.